memory: stop overriding unused `wcsdup()`/`_wcsdup()` system functions

vszakats opened the pull request:
Also ban them via `checksrc`.

The code continues to use `_tcsdup()`.

Assisted-by: Daniel Stenberg

Closes #17840

---

- [x] part of this PR moved to #17843

vszakats commented:
Into the rabbit hole:
76e047fc27b3a0b9e6d6d00cacf536e7b7c1b532 #7540
changed the initial value of `Curl_cwcsdup` from `_wcsdup` to `Curl_wcsdup`.
It did not apply the same change to `global_init()` a few lines below.
libtests also remains using `_wcsdup`.

Is this correct? Aren't all these supposed to be in sync?

bagder commented:
It looks like `_tcsdup()` is the only call left used in libcurl code, in four different places.

bagder commented:
> Except when assigning the function to Curl_cwcsdup. 

I was only referring to callers, users of this functionality.

vszakats commented:
Got it now!

So, deleted both macro redefs, since those are necessary for actual calls,
which do not exist.

Then fixed to use `Curl_wcsdup()` in global_init() (will move this to
separate PR), and changed the same in `libtests` to map to `_wcsdup`
instead of `wcsdup`.
